pwndbg/gdblib/kernel/__init__.py

```diff
@@ -135,6 +138,58 @@ def is_kaslr_enabled() -> bool:
     return "nokaslr" not in kcmdline()
 
 
+@pwndbg.lib.cache.cache_until("start")
+def kbase() -> int | None:
+    arch_name = pwndbg.gdblib.arch.name
+
+    address = 0
+
+    if arch_name == "x86-64":
+        address = get_idt_entries()[0].offset
+    elif arch_name == "aarch64":
+        address = pwndbg.gdblib.regs.vbar
```

Comment:
I don't think this method is going to be reliable... `vbar` will likely change often if you're running bootloaders or hypervisors. I think the manual search from before is preferable, maybe with this as a fallback.

Reply:
The problem with the manual search: The first opcode heavily depends on the kernel config for newer kernels without `_text`. For example, if `CONFIG_ARM64_BTI_KERNEL` is enabled the first instruction is a `bti c;` and if `CONFIG_DYNAMIC_FTRACE_WITH_CALL_OPS` is enabled the first few instructions look like this:
https://elixir.bootlin.com/linux/latest/source/arch/arm64/kernel/ftrace.c#L95
```
_stext+0: <ftrace_nop_ops> address
_stext+8: bti c;
_stext+12: mov x9, x30;
_stext+12: nop;
...
```
We could implement logic which tries to disassemble the bytes at `stext` to check if its an address or actual code and skip the first few config dependent instructions ?

But I think we should stick with the `vbar` as the main solution and add the manual search as a fallback, since it likely works for most use cases. Additionally, we could add checks to determine if the page is large enough (given that kernel text mappings are always quite large) and/or confirm it's the kernel base by searching for the Linux version string in the subsequent read-only page.

@gsingh93 
wdyt?

Reply:
Actually, I think we can just go with your current approach of just `vbar`. The scenarios I was worried about (hypervisors and secure monitors) will be setting `VBAR_EL2` and `VBAR_EL3`, `VBAR_EL1` will probably remain unchanged. `VBAR_EL1` may be incorrect while a bootloader is running, before the kernel is loaded, but I wouldn't be running `kbase` at that point anyway.

So let's go with what you have, and if it breaks any of my workflows we'll just figure it out then :)